src/NHibernate.Test/Async/Extralazy/ExtraLazyFixture.cs

```diff
@@ -34,13 +38,2207 @@ protected override string CacheConcurrencyStrategy
+				}
+
+				Assert.That(gavin.Companies.Count, Is.EqualTo(10));
+				Assert.That(Sfi.Statistics.FlushCount, Is.EqualTo(0));
+				Assert.That(Sfi.Statistics.PrepareStatementCount, Is.EqualTo(0));
+				Assert.That(NHibernateUtil.IsInitialized(gavin.Companies), Is.False);
+
+				// Test adding companies with IList interface
+				Sfi.Statistics.Clear();
+				for (var i = 10; i < 15; i++)
+				{
+					var item = new Company($"c{i}", i, gavin);
+					Assert.That(((IList)addedItems).Add(item), Is.EqualTo(i));
+					gavin.Companies.Add(item);
+				}
+
+				Assert.That(gavin.Companies.Count, Is.EqualTo(15));
+				Assert.That(Sfi.Statistics.FlushCount, Is.EqualTo(0));
+				Assert.That(Sfi.Statistics.PrepareStatementCount, Is.EqualTo(0));
+				Assert.That(NHibernateUtil.IsInitialized(gavin.Companies), Is.False);
+
+
+				// Check existance of added companies
+				Sfi.Statistics.Clear();
+				foreach (var item in addedItems.Skip(5))
+				{
+					Assert.That(gavin.Companies.Contains(item), Is.True);
+				}
+
+				Assert.That(Sfi.Statistics.FlushCount, Is.EqualTo(0));
+				Assert.That(Sfi.Statistics.PrepareStatementCount, Is.EqualTo(0));
+				Assert.That(NHibernateUtil.IsInitialized(gavin.Companies), Is.False);
+
+				// Check existance of not loaded companies
+				Assert.That(gavin.Companies.Contains(addedItems[0]), Is.True);
+				Assert.That(gavin.Companies.Contains(addedItems[1]), Is.True);
+
+				Assert.That(Sfi.Statistics.FlushCount, Is.EqualTo(0));
+				Assert.That(Sfi.Statistics.PrepareStatementCount, Is.EqualTo(2));
+				Assert.That(NHibernateUtil.IsInitialized(gavin.Companies), Is.False);
+
+				// Check existance of not existing companies
+				Sfi.Statistics.Clear();
+				Assert.That(gavin.Companies.Contains(new Company("test1", 15, gavin)), Is.False);
+				Assert.That(gavin.Companies.Contains(new Company("test2", 16, gavin)), Is.False);
+
+				Assert.That(Sfi.Statistics.FlushCount, Is.EqualTo(0));
+				Assert.That(Sfi.Statistics.PrepareStatementCount, Is.EqualTo(2));
+				Assert.That(NHibernateUtil.IsInitialized(gavin.Companies), Is.False);
+
+				if (initialize)
+				{
+					using (var e = gavin.Companies.GetEnumerator())
+					{
+						e.MoveNext();
+					}
+
+					Assert.That(NHibernateUtil.IsInitialized(gavin.Companies), Is.True);
+					Assert.That(gavin.Companies.Count, Is.EqualTo(15));
+				}
+
+				await (t.CommitAsync(cancellationToken));
+			}
+
+			using (var s = OpenSession())
+			using (var t = s.BeginTransaction())
+			{
+				gavin = await (s.GetAsync<User>("gavin", cancellationToken));
+				Assert.That(gavin.Companies.Count, Is.EqualTo(15));
+				Assert.That(NHibernateUtil.IsInitialized(gavin.Companies), Is.False);
+
+				await (t.CommitAsync(cancellationToken));
+			}
+		}
+
+		[TestCase(false, false)]
+		[TestCase(false, true)]
+		[TestCase(true, false)]
+		[TestCase(true, true)]
+		public async Task ListAddDuplicatedAsync(bool initialize, bool flush, CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
@maca88 can you please update AsyncGenerator's NUnit plugin not to generate cancellation token for `[TestCase]`?

Reply:
Done. Also updated the NHibernate configuration #2126.

Reply:
I just realised that this is because the test method lacks `[Theory]` attribute.